spk/rutorrent/src/service-setup.sh

```diff
@@ -0,0 +1,289 @@
+#!/bin/sh
+
+# Package
+PACKAGE="rutorrent"
+DNAME="ruTorrent"
+PACKAGE_NAME="com.synocommunity.packages.${PACKAGE}"
+
+# Others
+INSTALL_DIR="/usr/local/${PACKAGE}"
+WEB_DIR="/var/services/web"
+PATH="${INSTALL_DIR}/bin:${INSTALL_DIR}/usr/bin:${PATH}"
+APACHE_USER="$([ $(grep buildnumber /etc.defaults/VERSION | cut -d"\"" -f2) -ge 4418 ] && echo -n http || echo -n nobody)"
+BUILDNUMBER="$(/bin/get_key_value /etc.defaults/VERSION buildnumber)"
+
+GROUP="sc-download"
+GROUP_DESC="SynoCommunity's download related group"
+LEGACY_USER="rutorrent"
+LEGACY_GROUP="users"
+
+PYTHON_DIR="/usr/local/python3"
+VIRTUALENV="${PYTHON_DIR}/bin/virtualenv"
+
+# Sets recursive read / execute permissions for ${KEY} on specified directory
+# Usage: grant_basic_permissions "${SHARE_FOLDER}" "user:<user>"
+# Usage: grant_basic_permissions "${SHARE_FOLDER}" "group:<group>"
+grant_basic_permissions ()
```

Comment:
I needed this method because I'm in case where I need specific locations to be available for the web user to execute code... And I didn't want the apache user to be allowed to do *anything* in those directories.